dashboard/app/models/user.rb

```diff
@@ -830,6 +831,31 @@ def update_with_password(params, *options)
     end
   end
 
+  def update_primary_authentication_option(user: {email: nil, hashed_email: nil})
+    email = user[:email]
+    hashed_email = user[:hashed_email]
+
+    return false if email.nil? && hashed_email.nil?
+    return false if teacher? && email.nil?
+
+    # If an email option with a different email address already exists, destroy it
+    existing_email_option = authentication_options.find {|ao| ao.credential_type == 'email'}
+    existing_email_option&.destroy
+
+    # If an auth option exists with same email, set it to the user's primary authentication option
+    existing_auth_option = authentication_options.find {|ao| ao.email == email || ao.hashed_email == hashed_email}
+    if existing_auth_option
+      self.primary_authentication_option = existing_auth_option
+      return save
+    end
+
+    params = {credential_type: 'email', user: self}
+    params[:email] = email unless email.nil?
+    params[:hashed_email] = hashed_email if email.nil?
+    self.primary_authentication_option = AuthenticationOption.new(params)
```

Comment:
Will this actually save the new `AuthenticationOption` to the database? I'm not sure if saving the `User` will cascade to saving the `AuthenticationOption`

Reply:
yep! it [autosaves](http://guides.rubyonrails.org/association_basics.html#options-for-belongs-to-autosave) dependencies. although i can't find if/where we're explicitly setting autosave, so this i'll to verify in other environments besides dev because it could be different